kivy/core/audio_output/__init__.py

```diff
@@ -59,32 +108,128 @@
 from sys import float_info
 
 
+def _is_strict_mode():
+    '''Check if provider strict mode is enabled.
+
+    Returns True if KIVY_PROVIDER_STRICT is set to '1', 'true', or 'yes'
+    (case-insensitive).
+    '''
+    value = os.environ.get('KIVY_PROVIDER_STRICT', '').lower()
+    return value in ('1', 'true', 'yes')
+
+
 class SoundLoader:
     '''Load a sound, using the best loader for the given file type.
     '''
 
     _classes = []
+    _loaders_by_name = {}  # O(1) lookup by provider name
 
     @staticmethod
     def register(classobj):
         '''Register a new class to load the sound.'''
         Logger.debug('Audio: register %s' % classobj.__name__)
         SoundLoader._classes.append(classobj)
+        # Build _loaders_by_name for O(1) lookup
+        name = classobj.__name__
+        if name.startswith('Sound'):
+            name = name[len('Sound'):]
```

Comment:
Mmm, what about adding a class prop that exposes the name without guessing it via the class name?